File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Hardware/Claw.java
package org.firstinspires.ftc.teamcode.Hardware;

import com.qualcomm.robotcore.hardware.HardwareMap;
import com.qualcomm.robotcore.hardware.Servo;

public class Claw {
    private final Servo claw;

    public Claw(HardwareMap hardwareMap) {
        this.claw = hardwareMap.get(Servo.class, "claw");
    }

    double open = 0.8;
    double close = 0.5;
    public void open() {
        claw.setPosition(open);
    }

    public void close() {
        claw.setPosition(close);
    }

    public void setPosition(double position) {
        claw.setPosition(position);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/powerplay/subsystems/Claw.kt
package org.firstinspires.ftc.teamcode.powerplay.subsystems

import com.qualcomm.robotcore.hardware.HardwareMap
import com.qualcomm.robotcore.hardware.Servo
import org.firstinspires.ftc.teamcode.util.Globals
import org.firstinspires.ftc.teamcode.util.Subsystem

class Claw: Subsystem {
    lateinit var claw: Servo

    private enum class ClawStates(val pos: Double) {
        OPEN(Globals.CLAW_OPEN),
        CLOSED(Globals.CLAW_CLOSED),
    }

    private var clawState = ClawStates.OPEN

    fun openClaw() {
        clawState = ClawStates.OPEN
    }

    fun closeClaw() {
        clawState = ClawStates.CLOSED
    }

    override fun init (hardwareMap: HardwareMap) {
        claw = hardwareMap.servo["Claw"]
    }

    override fun update() {
        claw.position = clawState.pos
    }

    override fun reset() {
        openClaw()
    }
}

